### PR TITLE
Packaging update

### PR DESCRIPTION
**Pull request by LiamPattinson**
I've updated the project to use a modern build system and done some general reorganising. It should be ready to publish, though we may need to set a PyPI token first.

- Switched to pyproject.toml build system
- Updated files to match equivalents in latest freegs
- _version now autogenerated
- Moved tests to top-level directory
- Added black auto-commit workflow
- Added auto-publish workflow
- Updated workflow test versions
- General reformatting
- Updated README

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  release:
+    types: [published]
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
```
Comment (ZedThree) [inline .github/workflows/publish.yml]: I think there are more recent versions of these actions btw, like v3 and v4 respectively -- worth double-checking and bumping if necessary

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -10,12 +10,12 @@ on:
...
 
 jobs:
-  build:
+  build-and-test:
 
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.5, 3.6, 3.7, 3.8]
+        python-version: ["3.8", "3.9", "3.10"]
```
Comment (ZedThree) [inline .github/workflows/tests.yml]: Might as well add `"3.11"` too!

File: .github/workflows/publish.yml (hunk 1 above)
Comment (LiamPattinson) [reply .github/workflows/publish.yml]: Good call!

After looking through again, I spotted this in the black workflow:

```
    - uses: actions/checkout@v2
      with:
        ref: ${{ github.head_ref }}
```

It was copied over from a different project. Do you know if the `with` section is necessary?

Comment (LiamPattinson) [reply .github/workflows/publish.yml]: Never mind, found a better way to handle this

**Comment by LiamPattinson** [comment]
I found an official Black action. It doesn't auto-commit like the previous one, but it's much simpler and easy to maintain. Personally, I'd prefer to have the actions complain at me than to automatically commit, but I can revert to the previous workflow if you think that'd be better.

Edit: reverting back to the old version

**Review by ZedThree — APPROVED** [review]

**Pull request merged**